server/src/streem/util.ts

```diff
@@ -0,0 +1,33 @@
+import { Request } from 'express';
+import { createHmac, timingSafeEqual } from 'crypto';
+import { SIGNING_KEY_SECRET } from '../routes/webhooks';
+
+export function validateWebhookRequest(req: Request): Boolean {
+    const hmac = createHmac('sha256', Buffer.from(SIGNING_KEY_SECRET, 'utf-8').toString());
+
+    // make sure the request is current
+    const sentAt = req.headers['streem-sent-at']! as string;
+    const reqTimeSkew = Date.now() - Date.parse(sentAt)
+    const allowedSkew = 500 // adjust timedelta (in ms) as desired
+    if (reqTimeSkew > allowedSkew) {
+        return false; // request is too far in the past 
+    } else if (reqTimeSkew < -allowedSkew) {
+        return false; // request is in the future
+    }
+
+    // calculate the signature input message
+    const headerNames = (req.headers['streem-signature-headers']! as string).split(':')
+    const headers: string[] = headerNames.map(headerName => {
+        return `${headerName}=${req.headers[headerName.toLowerCase()]!}`;
+    });
+    const inputMsg = Buffer.from(`${headers.join(';')};`, 'utf-8').toString() + req.body;
```

Comment:
It looks like you're converting everything to strings before calculating the hash (i.e. the type of `inputMsg` is `string`, right?). While that might produce the same result in most cases, it's technically not correct since we define the signature to be over the binary representation of the body before any UTF-8 decoding has happened.

I think what you want is:

```ts
const inputMsg = Buffer.concat(Buffer.from(`${headers.join(';')};`, 'utf-8'), req.body);
```